doc/vector-extra/riscv-crypto-spec-vector-extra.adoc

```diff
@@ -0,0 +1,159 @@
+// Settings:
+:experimental:
+:reproducible:
+// needs to be changed? bug discussion started
+:WaveDromEditorApp: wavedrom-cli
+:imagesoutdir: images
+:icons: font
+:lang: en
+:listing-caption: Listing
+:sectnums:
+:toc: left
+:toclevels: 4
+:source-highlighter: pygments
+ifdef::backend-pdf[]
+:source-highlighter: coderay
+endif::[]
+:data-uri:
+:hide-uri-scheme:
+:stem: latexmath
+:footnote:
+:xrefstyle: short
+:bibtex-file: riscv-crypto-spec.bib
+:bibtex-order: alphabetical
+:bibtex-style: ieee
+
+//:This is the preamble.
+
+[colophon]
+= Colophon
+
+This document describes the Vector Cryptography Extra extensions to the
+RISC-V Instruction Set Architecture.
+
+This document is _Discussion Document_.
+Assume everything can change.
+This document is not complete yet and was created only for the purpose of conversation outside of the document.
+For more information, see link:http://riscv.org/spec-state[here].
+
+[NOTE]
+.Copyright and licensure:
+This work is licensed under a
+link:http://creativecommons.org/licenses/by/4.0/[Creative Commons Attribution 4.0 International License]
+
+[NOTE]
+.Document Version Information:
+====
+include::git-commit.adoc[]
+
+See link:https://github.com/riscv/riscv-crypto/doc/vector-extra[github.com/riscv/riscv-crypto/doc/vector-extra]
+for more information.
+====
+
+[acknowledgments]
+== Acknowledgments
+
+Contributors to this specification (in alphabetical order)
+include: +
+Ken Dockser,
+Markku-Juhani O. Saarinen,
+Nicolas Brunie,
+Richard Newell
+
+We are all very grateful to the many other people who have
+helped to improve this specification through their comments, reviews,
+feedback and questions.
+
+// ------------------------------------------------------------
+
+include::riscv-crypto-vector-extra-introduction.adoc[]
+
+// ------------------------------------------------------------
+
+<<<
+// ------------------------------------------------------------
+
+
+[[crypto_vector_extensions]]
+== Extensions Overview
+
+The section introduces all of the extensions in the Vector Cryptography Extra
+Instruction Set Extension Specification.
+
+
+All the Vector Crypto Extra Extensions can be built
+on _any_ embedded (Zve*) or application ("V") base Vector Extension.
+
+// See <<crypto-vector-element-groups>> for more details on vector element groups and the drawbacks of
+// small `VLEN` values.
+
+
+All _cryptography-specific_ instructions defined in this Vector Crypto specification (i.e., those
+in <<Zvkgs>>, but _not_ <<zvbc32e>>) shall
+be executed with data-independent execution latency as defined in the
+link:https://github.com/riscv/riscv-crypto/releases/tag/v1.0.1-scalar[RISC-V Scalar Cryptography Extensions specification].
+It is important to note that the Vector Crypto instructions are independent of the
+implementation of the `Zkt` extension and do not require that `Zkt` is implemented.
+
+//This specification includes a <<Zvkt>> extension that, when implemented, requires certain vector instructions
```

Comment:
Why doesn't Zvkt require that these new instructions be constant-time?

Reply:
`Zvkt` will require that these new extensions be constant time, I need to clarify this.
More precisely, Similarly to `Zvkg` which mandates

> To help avoid side-channel timing attacks, these instructions shall be implemented with data-independent timing.

 `Zvkgs` should inherit the same mandate and require constant-time even when `Zvkt` is not implemented.

Constant time for `Zvbc32e` will depend on `Zvkt`.